web/src/components/feature/FluidDialog.tsx

```diff
@@ -52,35 +52,24 @@ function convertComponentResponseToComponentInput(
   )
 }
 
-function normalizeArray(array: Array<number>): Array<number> {
-  // Normalize array if not "close enough" (|sum - 1| > 0.01) to 1
+function getNormalizationFactor(array: Array<number>): number {
   const arraySum: number = array.reduce(
     (partialSum, value) => partialSum + value,
     0
   )
-  return Math.abs(arraySum - 1) > 0.01
-    ? array.map((value) => value / arraySum)
-    : array
+  return Math.abs(arraySum - 1) > 0.01 ? arraySum : 1
 }
 
 function normalizeComponentComposition(
   componentComposition: TComponentComposition
 ): TComponentComposition {
-  // Normalize feedComponentRatios if necessary, else return
-  const normalizedValues: Array<number> = normalizeArray(
-    Object.values(componentComposition)
+  const factor = getNormalizationFactor(Object.values(componentComposition))
+  return Object.fromEntries(
```

Comment:
@mariuswinger I chose to create a new object instead of rewriting the existing one. This is best practice in React, since rewriting some part of an object or array will not trigger a rerender, while creating a new one that replaces the old, will. Not strictly necessary in this case, but good to be in the habbit

@StigNil An example of where using arrays would have been much smoother